mico-core/src/main/java/io/github/ust/mico/core/web/ApplicationController.java

```diff
@@ -139,12 +118,16 @@
             throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Application shortName or version does not match request body.");
         }
 
-        Optional<MicoApplication> applicationOptional = applicationRepository.findByShortNameAndVersion(shortName, version);
-        if (!applicationOptional.isPresent()) {
+        Optional<MicoApplication> existingApplicationOptional = applicationRepository.findByShortNameAndVersion(shortName, version);
+        if (!existingApplicationOptional.isPresent()) {
             throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Application '" + shortName + "' '" + version + "' was not found!");
         }
+        for (MicoService providedService : application.getServices()) {
```

Comment:
PUT endpoint should NOT update service list of existing application. The service list can already be manipulated with different endpoints so there is no need for put to also do this. And providing two ways of doing the same thing may be confusing. Also what would happen if the Provided application for this endpoint for some reason has no services?
In short: the PUT endpoint for applications should ignore the services completely!